pykusto/_src/pyspark_client.py

```diff
@@ -0,0 +1,114 @@
+from importlib.util import find_spec
+from typing import List, Iterable, Dict, Callable, Union, Tuple
+
+import pandas as pd
+from azure.kusto.data import ClientRequestProperties, KustoClient
+
+from pykusto import PyKustoClient, NO_RETRIES, KustoResponse, KQL, RetryConfig
+from .logger import _logger
+
+
+class DataframeBasedKustoResponse(KustoResponse):
+    """
+    In PySpark Kusto results are returned as dataframes. We wrap the dataframe with this object for compatibility with :class:`PyKustoClient`.
+    """
+    __dataframe: pd.DataFrame
+
+    # noinspection PyMissingConstructor
+    def __init__(self, dataframe: pd.DataFrame):
+        self.__dataframe = dataframe
+
+    def get_rows(self) -> List[Iterable]:
+        # noinspection PyTypeChecker
+        return self.__dataframe.values.tolist()
```

Comment:
it is recommended to use .to_numpy()  (see https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.values.html) but not critical